File: backend/GraficaModerna.Domain/Entities/CouponUsage.cs
namespace GraficaModerna.Domain.Entities;

public class CouponUsage
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string UserId { get; set; } = string.Empty;
    public string CouponCode { get; set; } = string.Empty;
    public Guid OrderId { get; set; }
    public DateTime UsedAt { get; set; } = DateTime.UtcNow;
}

File: app/Models/CouponUsage.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class CouponUsage extends Model
{
    use HasUuids;
    public $timestamps = false;

    protected $fillable = ['user_id', 'coupon_code', 'order_id', 'used_at'];
    
    protected $casts = ['used_at' => 'datetime'];
}